### PR TITLE
IA-2548 Enhancements and fix for the audit module

### PR DESCRIPTION
**Pull request by kemar**
Enhancements and fix for the audit module.

Related JIRA tickets : [IA-2548](https://bluesquare.atlassian.net/browse/IA-2548)

## Changes

- add type hints and comments to clarify how the audit module works
- add unit tests demonstrating some edge cases
- log empty modifications or modifications with only `updated_at`
- fix a bug in the groupsets API where the past version was not deep copied or serialized

## How to test

This is covered by unit tests.

[IA-2548]: https://bluesquare.atlassian.net/browse/IA-2548?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: hat/audit/tests.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        self.assertEqual(len(json_data), 1)
+
+        json_instance = json_data[0]
+        self.assertIsInstance(json_instance, dict)
+        self.assertEqual(json_instance["model"], "iaso.orgunit")
+        self.assertEqual(json_instance["pk"], self.org_unit.pk)
+        self.assertIsInstance(json_instance["fields"], dict)
+
+    def test_log_modification_with_nothing(self):
+        with self.assertLogs(logger=audit_models.logger, level="WARNING") as caught_msg:
```
Comment (tdethier) [inline hat/audit/tests.py:42]: Interesting, I had never used `self.assertLogs`

File: iaso/tests/api/test_org_units_bulk_update.py [hunk 2]
```diff
@@ -225,6 +225,8 @@ def test_org_unit_bulkupdate_select_some(self):
     def test_org_unit_bulkupdate_select_all(self):
         """POST /orgunits/bulkupdate happy path (select all)"""
 
+        m.OrgUnit.objects.update(validation_status=m.OrgUnit.VALIDATION_NEW)
```
Comment (tdethier) [inline iaso/tests/api/test_org_units_bulk_update.py:228]: Good catch :+1:

**Review by tdethier — APPROVED** [review]
LGTM :+1: 

There's still a lot to do to improve auditing, but that's a nice step in the right direction.
